Ignore Initial Unprotected Uppercase In Translated Titles

michalfarago opened the pull request:
Fixes #10459

What was done:
---
Although not necessary, I've replaced the old while loop with following:
![image](https://github.com/JabRef/jabref/assets/85625273/589d1ae2-2161-47d8-a4da-db3c62ee9a84)

Since in test class there are two types of checkers:
![image](https://github.com/JabRef/jabref/assets/85625273/fc9a0887-f5fb-4b0a-be68-3956b97f80c7)
I've separated the test to nested classes, so it's easier to see, which tests are related to which checker + the tests for valid/invalid title were grouped using @ParameterizedTest.
![image](https://github.com/JabRef/jabref/assets/85625273/f5ee6734-6470-4d69-b13b-d2d2054144d6)


The message {0} is just a placeholder which contains previous test method name, to indicate user what does the test check.
The old tests were extended to also verify correct verification of the _Title_ is in square brackets (translated title).
![image](https://github.com/JabRef/jabref/assets/85625273/e5c2d86a-0a07-4463-81cd-1c0c20e6f858) 

<!-- 
Describe the changes you have made here: what, why, ... 
Link issues that are fixed, e.g. "Fixes #333".
If you fixed a koppor issue, link it, e.g. "Fixes https://github.com/koppor/jabref/issues/47".
The title of the PR must not reference an issue, because GitHub does not support autolinking there.
-->

<!-- 
- Go through the list below. Please don't remove any items.
- [x] done; [ ] not done / not applicable
-->

### Mandatory checks
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

calixtus commented:
Can you please convert the tests to parameterized tests. They look jast as made for that 😉

michalfarago commented:
> Can you please convert the tests to parameterized tests. They look jast as made for that 😉

Sure thing! I've rewritten them to use parameterized tests. I've used MethodSource, but let me know if you'd prefer to migrate them to a CSV file and use that instead (as there aren't many, I didn't find it necessary).

I've also noticed that there were some duplicate tests, where the tested title was the same, the operation was the same (asserEquals), but the method name was different.  I've removed those duplicates as they check the same thing.

I've also kept the previous method names in the test display to indicate what the test checks, but let me know if I should only keep the titles.
